x-pack/plugins/apm/server/lib/transactions/get_transaction/index.ts

```diff
@@ -37,8 +37,8 @@ export function getTransaction({
           bool: {
             filter: asMutableArray([
               { term: { [TRANSACTION_ID]: transactionId } },
-              { term: { [TRACE_ID]: traceId } },
-              ...rangeQuery(start, end),
+              ...(traceId ? [{ term: { [TRACE_ID]: traceId } }] : []),
+              ...('start' in setup ? rangeQuery(setup.start, setup.end) : []),
```

Comment:
Are `setup.start` and `setup.end` typed as optionals here? Would be great if they were but I seem to recall we require them on every api

Reply:
(or maybe it's the opposite problem I'm thinking about where they are always optional, even when they are required on the api)

Reply:
This function is used in different endpoints, one that does have a start/end query parameter, and one that doesn't. The type for `setup` here is a union type of setup without start/end (not defined) and setup with start end (always defined).